optimum/exporters/openvino/model_patcher.py

```diff
@@ -1493,16 +1493,51 @@ def _phi3_self_attn_sdpa_forward(
     return attn_output, None, past_key_value
 
 
+def select_ext_factor(
+    seq_len: torch.Tensor, max_pos_embeddings: torch.Tensor, short_factor: torch.Tensor, long_factor: torch.Tensor
+):
+    return torch.where(seq_len <= max_pos_embeddings, short_factor, long_factor)
+
+
+def long_rope(self, x, position_ids, seq_len=None):
+    seq_len = torch.max(position_ids) + 1
+    original_max_position_embeddings = (
+        self.original_max_position_embeddings
+        if hasattr(self, "original_max_positional_embeddings")
+        else self.config.original_max_position_embeddings
+    )
+    max_position_embeddings = (
+        self.max_position_embeddings
+        if hasattr(self, "max_position_embeddings")
+        else self.config.max_position_embeddings
+    )
+    inv_freq = select_ext_factor(seq_len, original_max_position_embeddings, self.inv_freq, self.long_inv_freq)
```

Comment:
Let us add a comment:

`slow down all frequencies by scale factor for long prompts that makes attention more stable, i.e. preserve model accuracy`